tusclient/uploader.py

```diff
@@ -211,7 +223,35 @@ def upload_chunk(self):
         """
         Upload chunk of file.
         """
+        self._retried = 0
         self._do_request()
         self.offset = int(self.request.response_headers.get('upload-offset'))
         msg = '{} bytes uploaded ...'.format(self.offset)
         print(msg)
+
+    def _do_request(self):
+        # TODO: Maybe the request should not be re-created everytime.
+        #      The request handle could be left open until upload is done instead.
+        self.request = TusRequest(self)
+        try:
+            self.request.perform()
+            self.verify_upload()
+        except TusUploadFailed as error:
+            self.request.close()
+            self._retry_or_cry(error)
+        finally:
+            self.request.close()
+
+    def _retry_or_cry(self, error):
+        if self.retries > self._retried:
+            time.sleep(self.retry_delay)
```

Comment:
Since I don't have much experience with Python, I can't really help there. But let's not delay this PR because of this. If a users wants to interrupt an upload, we will come up with a solution :)